### PR TITLE
Add methods to recalculate and toggle static geometry

### PR DESCRIPTION
**Pull request by levidavidmurray**
`SteamAudioGeometry` wasn't working properly for my procedurally generated geometry as my actual geometry is being globally positioned after `SteamAudioGeometry::_ready` is called. So I needed a way to "recalculate" the static audio geometry.

I figured I'd also add an easy way to enable and disable the audio geometry.

https://github.com/stechyo/godot-steam-audio/assets/46084870/cb2d45b3-1ac6-40f1-848b-b6989899cd95

I had some issues getting the GDExtension to build properly on Windows (msvc).
`scons platform=windows target=template_debug dev_build=true` built and worked fine. Omitting the `dev_build=true` flag builds fine, but the Godot project would instantly crash.

I did notice quite a size difference between my dll (~500KB) and the one provided in `project/addons` (~1.8MB). I tried building with `use_mingw=true` and after about 30-40 minutes it finally finished (subsequent builds weren't much faster). In any case, it did end up producing a dll around the same size and the GDExtension worked fine. The build time is quite the hindrance though. I'll probably find a solution with a fresh pair of eyes tomorrow (likely related: https://github.com/godotengine/godot/issues/30892).

### CODE REVIEW

File: project/addons/godot-steam-audio/bin/libgodot-steam-audio.windows.template_debug.x86_64.dll
Comment (stechyo) [inline project/addons/godot-steam-audio/bin/libgodot-steam-audio.windows.template_debug.x86_64.dll:1]: Just leaving a note here to block the merge until we fix this: #33 removes this DLL and fixes the .gitignore to include .dlls and .sos on this folder, so after that's merged we can remove this

File: src/geometry.cpp [hunk 1]
```diff
@@ -175,15 +174,66 @@ void SteamAudioGeometry::create_meshes_from_coll_inst_3d(CollisionShape3D *coll_
...
+
+void SteamAudioGeometry::register_geometry() {
+	for (auto ipl_mesh : meshes) {
+		SteamAudioServer::get_singleton()->add_static_mesh(ipl_mesh);
+	}
+}
+
+void SteamAudioGeometry::unregister_geometry() {
+    for (auto mesh : meshes) {
+        iplStaticMeshRemove(mesh, SteamAudioServer::get_singleton()->get_global_state()->scene);
```
Comment (stechyo) [inline src/geometry.cpp]: Nitpick: could you change this to be more consistent with register_geometry() above? New remove_static_mesh method on SteamAudioServer and all that. Thanks!

File: src/geometry.hpp [hunk 2]
```diff
@@ -17,19 +17,25 @@ class SteamAudioGeometry : public Node3D {
...
+	bool disabled = false;
+
 	SteamAudioGeometry();
 	~SteamAudioGeometry();
 	void _ready() override;
 
+	void recalculate();
 	Ref<SteamAudioMaterial> get_material();
 	void set_material(Ref<SteamAudioMaterial> p_material);
+	bool get_disabled() const { return disabled; }
```
Comment (stechyo) [inline src/geometry.hpp]:
```suggestion
	bool is_disabled() const { return disabled; }
```

**Comment by stechyo** [comment]
Hi, thanks for this! I hadn't thought of this use-case and your code looks good (just had some nitpicks).  
I'll try to test your PR this week, but I'll only merge it after #33 and after releasing 0.1.0. I did not want to introduce any more changes in this release, hope you understand.

Semi-unrelated things:
Omitting `dev_build=true` creates a different binary name, so it might be the reason for your crashing.  
I'm building the Windows binary with mingw mostly because I don't use Windows except for testing this extension's releases. I wasn't aware that MSVC binaries were that much smaller, so I'll probably make a Windows/MSVC step when I get around to doing #31. Just a check, size and compilation time is the only difference between MSVC and MinGW builds, right?  
Feel free to keep building on MSVC for now, since there doesn't seem to be a difference.

**Comment by levidavidmurray** [comment]
> I did not want to introduce any more changes in this release, hope you understand

Totally fine! I'm more than okay with running my own builds for the time being.

> Just a check, size and compilation time is the only difference between MSVC and MinGW builds, right?

I couldn't get MSVC builds to run without `dev_build=true`. I'm quite certain it's more than just naming as I did ensure the binary paths matched up in `libgodot-steam-audio.gdextension`.

**Comment by stechyo** [comment]
Is it possible that it has something to do with the fact that my fork of Godot is compiled on MinGW?

(Will test your changes this weekend and merge if all the review comments are solved and it looks good. Great contrib, thanks!)

**Comment by levidavidmurray** [comment]
> (Will test your changes this weekend and merge if all the review comments are solved and it looks good. Great contrib, thanks!)

Did you leave review comments? Missing repo setting?

https://github.com/stechyo/godot-steam-audio/assets/46084870/f2216c23-0436-4fc1-a162-10c2d17a6715

**Comment by stechyo** [comment]
... sorry, new to having other people's PRs on my own repos. I think they should show up now.

**Comment by levidavidmurray** [comment]
All good! 😅 And yup, I see them now. I'm currently in the depths of trying to get pathing simulations working. I was having a hard time getting small closed off interiors to simulate how I want them to (I tried many different configurations).

In any case, I'll get around to the PR comments either when I get the bare minimum path sim working, or I give up.

File: project/addons/godot-steam-audio/bin/libgodot-steam-audio.windows.template_debug.x86_64.dll
Comment (levidavidmurray) [reply project/addons/godot-steam-audio/bin/libgodot-steam-audio.windows.template_debug.x86_64.dll:1]: Done. I also added a few more build files (possibly specific to msvc) to the .gitignore

File: src/geometry.cpp (hunk 1 above)
Comment (levidavidmurray) [reply src/geometry.cpp]: Done. I wasn't sure if you were expecting any meshes removed via `remove_static_mesh` to be removed from `meshes_to_add` in the same fashion as `add_static_mesh` (if global state uninitialized). I'm not exactly sure how likely this is to happen, but opted for it anyway.

File: src/geometry.hpp (hunk 2 above)
Comment (levidavidmurray) [reply src/geometry.hpp]: Done!

File: src/geometry.cpp (hunk 1 above)
Comment (stechyo) [reply src/geometry.cpp]: Hm yeah it's extremely unlikely but still better to keep it there and cover that use case anyway, thanks!

**Comment by stechyo** [comment]
Just tested, works fine and no crashes. Merging, thanks!

**Pull request merged**